pagerduty_on_call/impl.py

```diff
@@ -0,0 +1,29 @@
+from sym.sdk.annotations import hook, reducer
+from sym.sdk.exceptions import SymException
+from sym.sdk.integrations import pagerduty, slack
+from sym.sdk.templates import ApprovalTemplate
+
+
+# Reducers fill in the blanks that your workflow needs in order to run.
+@reducer
+def get_approvers(event):
+    """Route Sym requests to a channel specified in the sym_flow."""
+
+    # allow_self lets the requester approve themself, which is great for testing!
+    return slack.channel("#sym-requests", allow_self=True)
+
+
+@hook
+def on_request(event):
+    """If the requester is on-call, auto-approve their requests"""
+    try:
+        if pagerduty.is_on_call(event.user):
+            original_reason = event.payload.fields.get("reason", "")
```

Comment:
`reason` is a required field so we should just access it; this suggests it may not exist which may trip someone up.

```suggestion
            original_reason = event.payload.fields["reason"]
```